accepted/2020/workloads/workload-resolvers.md

```diff
@@ -0,0 +1,237 @@
+
+The multiplexing between different versions of an OS SDK can be implemented in the `WorkloadManifest.targets` file that is part of the workload manifest, or other .targets it imports.  For example, the `Xamarin.Manifest.iOS.Framework.targets` file imported by `WorkloadManifest.targets` in the previous example could look like this:
+
+```xml
+<Project>
+    <Import Project="Sdk.targets" Sdk="Xamarin.iOS.Sdk" Condition="$([MSBuild]::VersionLessThan($(TargetPlatformVersion), '15.0'))" />
+    <Import Project="Sdk.targets" Sdk="Xamarin.iOS.Sdk.15.0" Condition="$([MSBuild]::VersionGreaterThanOrEquals($(TargetPlatformVersion), '15.0'))" />
+</Project>
+```
+
+This would import the iOS targets from the Xamarin.iOS.Sdk.15.0 workload pack if the `TargetPlatformVersion` is 15.0 or higher, and from the Xamarin.iOS.Sdk workload pack otherwise.
+
+### Workload props files
+
+Workloads may also provide MSBuild logic in an `AutoImport.props` file in the Sdk folder of the workload pack which will be imported before the body of the project file.  We expect this to be use for default globs, for example to specify that all .xaml files should be included as `Page` items.
+
+However, `AutoImport.props` files from workloads are subject to restrictions and must be carefully authored.  This is because since they need to be imported before the body of the project is evaluated, they can't be included conditionally based on the properties in the project that define whether the workload is in use.  So the `AutoImport.props` files from all workload packs will be imported for all projects using Microsoft.NET.Sdk.
+
+MSBuild uses multiple phases for evaluation, and properties and imports are evaluated in a phase before items.  This means it is possible to have conditions on item or `ItemGroup` elements in a .props file that depend on properties that are defined in the body of the project or in .targets files.
+
+So workloads may include MSBuild items in the `AutoImport.props` file if they are appropriately conditioned to only activate when the workload is in use.  They should not set MSBuild properties, as there is no way to set those properties only if the workload is in use.  If workload wants to set a default value for a property, then it should do so in its .targets file with a condition to set the property if it is not already set.
+
+Note that in contrast to the `WorkloadManifest.targets` file, which is part of the workload manifest package, the `AutoImport.props` files are part of the workload packs (under the Sdk folder).  This is so that they can be imported only if the workload is installed, rather than imported for all workloads whether they are installed or not.  This reduces the impact of a bug in an `AutoImport.props` file, and means that if there is an issue in the .props file, it can be worked around by uninstalling the workload.
```

Comment:
This paragraph and the one before it seems like they duplicate some of the same content and could be streamlined a bit.